controllers/capabilities/backend_controller.go

```diff
@@ -74,12 +76,57 @@ func (r *BackendReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		reqLogger.V(1).Info(string(jsonData))
 	}
 
+	// Ignore deleted Backends, this can happen when foregroundDeletion is enabled
+	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
+	if backend.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(backend, backendFinalizer) {
+		err = controllerhelper.ReconcileFinalizers(backend, r.Client(), backendFinalizer)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		err = r.removeBackend(backend)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	}
+
 	// Ignore deleted Backends, this can happen when foregroundDeletion is enabled
 	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
 	if backend.DeletionTimestamp != nil {
 		return ctrl.Result{}, nil
 	}
 
+	if !controllerutil.ContainsFinalizer(backend, backendFinalizer) {
+		err = controllerhelper.ReconcileFinalizers(backend, r.Client(), backendFinalizer)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if backend.Spec.ProviderAccountRef != nil {
+		providerAccount, err := controllerhelper.LookupProviderAccount(r.Client(), backend.GetNamespace(), backend.Spec.ProviderAccountRef, r.Logger())
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// Retrieve ownersReference of tenant CR that owns the Backend CR
+		tenantCR, err := controllerhelper.RetrieveTenantCR(providerAccount, r.Client())
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// If tenant CR is found, set it's ownersReference as ownerReference in the BackendCR CR
+		if tenantCR != nil {
+			err := controllerhelper.SetOwnersReference(backend, r.Client(), tenantCR)
```

Comment:
Let's add owner references in another PR. IMO out of scope.

Besides:
* Product controller does not add them
* controllerhelper.SetOwnersReference does not need to be implemented. You can call `r.SetOwnerReference(tenantCR, backend)`
* The method `RetrieveTenantCR` has several issues
  * the tenant name parsed from the URL and the `OrganizationName` are case sensitive and might not match. 
  * `OrganizationName`  is not unique and the selected tenant owner might not be the actual tenant owning the backend
  * What I would do is list all the tenants in the same namespace as the backend (or product) and read the secret from the `tenantSecretRef` field. When a secret is found which`adminURL` matches the backend's provider account URL, then you have the tenant owning the backend resource.

Reply:
Sorry, also, good point in pushing it in another PR. Going to move it to another ticket.